granularflow.go

```diff
@@ -190,9 +197,15 @@ func (c *baseConn) describe(r *buff.Reader, q *gfQuery) (descPair, error) {
 			// input descriptor
 			descs.in = descriptor.Pop(r.PopSlice(r.PopUint32()))
 
-			// output descriptor
-			r.Discard(16) // descriptor ID
-			descs.out = descriptor.Pop(r.PopSlice(r.PopUint32()))
+			// output descriptor ID
+			outID := r.PopUUID()
+
+			if outID == descriptor.NilID {
```

Comment:
Hm, I'm not sure I understand how this check works... outID comes from our protocol, right?

Reply:
correct. from the [docs](https://www.edgedb.com/docs/internals/protocol/typedesc#type-descriptors):

> There is one special type with type id of zero: 00000000-0000-0000-0000-000000000000. The describe result of this type contains zero blocks. It’s used when a statement returns no meaningful results, e.g. the CREATE DATABASE example statement.

Reply:
maybe I should call it `descriptor.ZeroID` :thinking: then `nilDecoder` would become `zeroDecoder`... or something.

Reply:
Ah, I get it now, thanks